### PR TITLE
Add stubs for werkzeug

### PR DESCRIPTION
**Pull request by rowillia**

### CODE REVIEW

**Comment by matthiaskramm** [comment]
I see that you're also doing some `stdlib/` changes in your PR. Could you split those into another pull request?

File: stdlib/2.7/collections.pyi [hunk 1]
```diff
@@ -7,8 +7,9 @@
 # NOTE: These are incomplete!
 
 from typing import (
-    Any, Dict, Generic, TypeVar, Iterable, Tuple, Callable, Mapping, overload, Iterator,
-    Sized, Optional, List, Set, Sequence, Union, Reversible, MutableMapping, MutableSequence
+    Any, Container, Dict, Generic, TypeVar, Iterable, Tuple, Callable, Mapping, overload,
```
Comment (gvanrossum) [inline stdlib/2.7/collections.pyi]: This looks like it's just a cleanup. Can you roll that back? It's not pertinent to the purpose of the PR, and in general we don't accept cleanup diffs (they just cause churn without much value).

File: stdlib/2.7/cookielib.pyi [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Stubs for cookielib (Python 2)
```
Comment (gvanrossum) [inline stdlib/2.7/cookielib.pyi]: Can you make this a separate PR? It's easier to manage things that way.

File: stdlib/3/collections/__init__.pyi [hunk 3]
```diff
@@ -13,9 +13,11 @@ from typing import (
 # These are exported.
 # TODO reexport more.
 from typing import (
+    Container as Container,
```
Comment (gvanrossum) [inline stdlib/3/collections/__init__.pyi]: Good catch, but I'd like to see this (and anything else that's not strictly part of werkzeug) in a separate PR.

**Comment by gvanrossum** [comment]
Hey Roy, are you still interested in this?

**Comment by gvanrossum** [comment]
Hey @rowillia are you interested in bringing this to completion? I've cherry-picked your addition of cookielib and your changes to collections and urllib. Also, one reason for the merge conflict is probably that I renamed the `2.7` directories to `2`. It *may* be easier to just copy all the files to a backup tree and start over with a new PR. Or perhaps all you need to do is move the werkzeug stubs.

**Comment by rowillia** [comment]
@gvanrossum Rebasing this now.  I need https://github.com/python/typeshed/pull/691 in to get this passing in mypy/2.7

**Comment by gvanrossum** [comment]
Thanks. I need to take a break from all work, hopefully it can wait or someone else will be able to review it.

**Comment by rowillia** [comment]
@gvanrossum Enjoy your time off!

**Comment by rowillia** [comment]
@gvanrossum Rebased with passing tests now that the `MutableSet` PR is in.  Hope you enjoyed your time unplugging 😄

**Comment by gvanrossum** [comment]
Thanks! Is your next project going to be flask stubs?

**Comment by rowillia** [comment]
@gvanrossum it will be 😄

**Comment by gvanrossum** [comment]
BTW, this puts the Python 2 files in thirdparty/2.7/ -- they should be in thirdparty/2/. See https://github.com/python/typeshed/pull/635 for why. I will fix this in https://github.com/python/typeshed/pull/717.
